### DISCUSSION TITLE
Channel naming

### DISCUSSION THREAD

**Discussion by ddennetiere** [PyMoDAQ/PyMoDAQ#305] [Q&A]
Dear PyMoDAQ community,
I am currently running pymodaq 3.5.6 (I know...).
I have a sensor that reads the temperature at several points in my setup.
I have set it up as a multi channel daq_viewer 0D, since in my opinion it keeps the data a bit cleaner than N mono channel daq_viewer.
Each channel has a unique ID sent using labels:

`data_tot, labels = self.controller.get_temperature()`

`self.data_grabed_signal_temp.emit([DataFromPlugins(name='PTC Bricklets', data=data_tot, dim='Data0D', labels=labels)])`

That architecture works well and I get nice plots with the IDs as legends in pyqtgraph. 

The issue arrises when I do a scan: in the h5browser, I have my data but each channel is now named CH000, CH001, etc. And the ID is nowhere to be found. Since each sensor is independent and precisely located in my experience, it is a problem, I need to know which is which...

Is there something I am missing in the way I send my data ? Could the ID appear somewhere in the metadata in the h5 file ?

The answer may be to simply upgrade to pymodaq 4.2.0, but that will be a costly operation and I would rather not risk it if there is a workaround. 

Thank you for any help you could provide,
David

**Comment by seb5g** [comment]
Did you look into the metadata of the data nodes? I don't quite remember if I saved at that time the labels (but it should...). Moreover, the channels should be saved in the same order they have been emitted (within the list). 

Why would it be costly to move to 4.x? Do you have many custom plugins? The architecture is not that much different... especially if you start from a fresh template!

**Reply by ddennetiere** [reply]
Thank you for the fast answer!
I have looked for the ID in the metadata of the node and haven't found it in the title nor name for example where I would have expected to find them.
There is no way I know the order they're emmited in, as the data is handled through a dictionary. And there is a need for modularity, hence a dictionary. But as a last resort i can refactor this in a way that shows order in the detector settings I guess. The challenge being that this is a metrology setup and that the data should be self sufficient for archiving. 

As for 4.x, I did try to install it and encountered many hurdles (bocs dependency, had to fix two bugs in daq viewer and dashboard, for some reason pymodaq didn't find pytables which is in its conda environment...), but i finally overcame enough to at least try it out and then no way to install my 6 custom plugins, so i would have to refactor a great deal. Which is not to say I won't, just not immediately.
